ui/apps/platform/cypress/integration/collections/collectionsCrudWorkflow.test.js

```diff
@@ -0,0 +1,150 @@
+    const auth = { bearer: Cypress.env('ROX_AUTH_TOKEN') };
+
+    cy.request({
+        url: `${collectionsApi.baseUrl}?query.query=Collection Name:"${collectionName}"`,
+        auth,
+    }).as('listCollections');
+
+    cy.get('@listCollections').then((res) => {
+        const collection = res.body.collections.find(({ name }) => name === collectionName);
+        if (collection) {
+            const { id } = collection;
+            const url = `${collectionsApi.baseUrl}/${id}`;
+            cy.request({ url, auth, method: 'DELETE' });
+        }
+    });
+}
+
+describe('Create collection', () => {
+    withAuth();
+
+    beforeEach(function beforeHook() {
+        if (!hasFeatureFlag('ROX_OBJECT_COLLECTIONS')) {
+            this.skip();
+        }
+        // Ignore autocomplete requests
+        // TODO Remove this once the feature is in
+        cy.intercept(`${collectionsApi.autocomplete}`, {});
+    });
+
+    const collectionName = 'Financial deployments';
+    const clonedName = `${collectionName} (COPY)`;
+
+    it('should allow creation of a new collection', () => {
+        // Cleanup from potential previous test runs
+        tryDeleteCollection(collectionName);
+        tryDeleteCollection(clonedName);
+
+        visitCollections();
+
+        cy.get('a:contains("Create collection")').click();
+        cy.get('input[name="name"]').type(collectionName);
+        cy.get('input[name="description"]').type('A collection for financial data');
+
+        cy.get('button:contains("All deployments")').click();
+        cy.get('button:contains("Deployments with labels matching")').click();
+        cy.get('input[aria-label="Select label key for deployment rule 1 of 1"]').type('meta/name');
+        cy.get(`button:contains('Add "meta/name"')`).click();
+        cy.get('input[aria-label="Select label value 1 of 1 for deployment rule 1 of 1"]').type(
+            'visa.*'
+        );
+        cy.get(`button:contains('Add "visa.*"')`).click();
+        cy.get('button[aria-label="Add deployment label value for rule 1"]').click();
+        cy.get('input[aria-label="Select label value 2 of 2 for deployment rule 1 of 1"]').type(
+            'mastercard.*'
+        );
+        cy.get(`button:contains('Add "mastercard.*"')`).click();
+
+        cy.get('button:contains("All namespaces")').click();
+        cy.get('button:contains("Namespaces with names matching")').click();
+        cy.get('input[aria-label="Select value 1 of 1 for the namespace name"]').type('payments');
+        cy.get(`button:contains('Add "payments"')`).click();
+
+        cy.get('button:contains("All clusters")').click();
+        cy.get('button:contains("Clusters with names matching")').click();
+        cy.get('input[aria-label="Select value 1 of 1 for the cluster name"]').type('production');
+        cy.get(`button:contains('Add "production"')`).click();
+
+        cy.get('button:contains("Save")').click();
+
+        cy.get('a:contains("Financial deployments")');
```

Comment:
What are pro and con for understandability to add table cell to selector?

```js
cy.get('td[data-label="Collection"] a:contains("Financial deployments")');
```

Reply:
A separate stylistic question which I ask but have no strong opinion, whether to use template literal:

```js
cy.get(`a:contains("${collectionName}")`);
```

Reply:
Aha, I meant to change the second one before I pushed and looks like it slipped by.